hd/hd.c

```diff
@@ -0,0 +1,91 @@
+#include <stdio.h>
+#include <ctype.h>
+
+int t(int s, int c);
+
+int main(void)
+{
+  int s = 0;
+  int c;
+  while ((c = getchar()) != EOF) {
+    s = t(s, c);
+    if (s < 0) return s;
+    if (s == 5) {
+      printf("# ");
+    } else if (s == 9) {
+      printf(" #");
+    }
+    printf("%c", c);
+  }
+  return 0;
+}
+
+int t(int s, int c) {
+  switch (s) {
+    case 0: {
+      switch (c) {
+        case '#': return 1;
+        case '\r': return 4;
+        case '\n': return 0;
+        default: return 5;
+      }
+    }
+    case 1: {
+      switch (c) {
+        case ' ': return 2;
+        default: return 9;
+      }
+    }
+    case 2: {
+      switch (c) {
+        case '\r': return 3;
+        case '\n': return 0;
+        default: return 2;
+      }
+    }
+    case 3: {
+      switch (c) {
+        case '\n': return 0;
+        default: return 2;
+      }
+    }
+    case 4: {
+      return 0;
+    }
+    case 5: {
+      switch (c) {
+        case '\r': return 6;
+        case '\n': return 0;
+        default: return 7;
+      }
+    }
+    case 6: {
+      switch (c) {
+        case '\n': return 0;
+        default: return 5;
+      }
+    }
+    case 7: {
+      switch (c) {
+        case '\r': return 8;
+        case '\n': return 0;
+        default: return 7;
+      }
+    }
+    case 8: {
+      switch (c) {
+        case '\n': return 0;
+        default: return 7;
+      }
+    }
+    case 9: {
+      switch (c) {
+        case '\r': return 8;
+        case '\n': return 0;
+        default: return 7;
+      }
+    }
+  }
+  fprintf(stderr, "[WARN] unexpected case");
+  return -1;
```

Comment:
So if this function returns -1, then this return value triggers `if (s < 0) return s;` in turn? If so, I think it would be nice if we could terminate the program with `exit(EXIT_FAILURE)`, since I think it would give it a more semantics. Though the exit code for abnormal termination is any integer other than 0, as far as I know.

Reply:
LGTM. Thanks for your help 🙏